src/modules/Auth/pages/RecoverPassword/RecoverPassword.tsx

```diff
@@ -0,0 +1,62 @@
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { Trans } from 'react-i18next';
+import { useEffect, useState } from 'react';
+import { Box } from '@mui/material';
+
+import { RecoverForm } from 'modules/Auth/features/RecoverPassword';
+import { recoveryLinkHealthCheckApi } from 'api';
+import { StyledBodyLarge, StyledLinkBtn } from 'shared/styles';
+import { Spinner } from 'shared/components';
+import { page } from 'resources';
+
+export const RecoverPassword = () => {
+  const [searchParams] = useSearchParams();
+
+  const navigate = useNavigate();
+
+  const key = searchParams.get('key');
+  const email = searchParams.get('email');
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [error, setError] = useState<Error>();
+
+  useEffect(() => {
+    recoveryLinkHealthCheckApi({ email, key })
+      .catch((error) => {
+        if (error instanceof Error) {
+          setError(error);
+        }
+      })
+      .finally(() => setIsLoading(false));
+  }, [email, key]);
+
+  if (isLoading) {
+    return (
+      <Box display="flex" flex={1} justifyContent="center" alignItems="center" textAlign="center">
+        <Spinner />
+      </Box>
+    );
+  }
+
+  if (error) {
+    return (
+      <Box display="flex" flex={1} justifyContent="center" alignItems="center" textAlign="center">
```

Comment:
Maybe It's time to discuss the styling of components, but so far we stick the usage of Styled<Component> as common approach like you applied in [RecoverForm.styles.ts](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1510/files#diff-285908779dbf07e580c21b701a9afedcceb52cdb7c6d1c809e5c54cdb31cd4ca)

Reply:
I opted to use inline styles here because the `RecoverPassword` component is a page, and so far I've only seen the styled component approach used for components in the `features` directory. However, I agree that these styles could be abstracted. Do you have an opinion on where that file could go?

Reply:
Yeap, make sense in this particular case.